src/aks-preview/azext_aks_preview/agentpool_decorator.py

```diff
@@ -1070,6 +1072,35 @@ def set_up_virtual_machines_profile(self, agentpool: AgentPool) -> AgentPool:
 
         return agentpool
 
+    def set_up_managed_system_mode(self, agentpool: AgentPool) -> AgentPool:
+        """Handle the special ManagedSystem mode by resetting all properties except name and mode.
+
+        :param agentpool: the AgentPool object
+        :return: the AgentPool object
+        """
+        mode = self.context.raw_param.get("mode")
+        if mode == CONST_NODEPOOL_MODE_MANAGEDSYSTEM:
+            # Raise error if agentpool is None
+            if agentpool is None:
+                raise CLIInternalError("agentpool cannot be None for ManagedSystem mode")
+
+            # Instead of creating a new instance, modify the existing one
+            # Keep name and set mode to ManagedSystem
+            name = agentpool.name
+            agentpool.mode = CONST_NODEPOOL_MODE_MANAGEDSYSTEM
+
+            # Make sure all other attributes are None
+            for attr in vars(agentpool):
+                if attr != 'name' and attr != 'mode' and not attr.startswith('_'):
+                    if hasattr(agentpool, attr):
+                        setattr(agentpool, attr, None)
```

Comment:
what about init a new agentpool model and only set name and mode? But there might be some default values for some fields, not sure if there's a need to clear them or not.

Reply:
there's default field need to clear out